scripts/workarounds/boot-order/metal-lib.sh

```diff
@@ -0,0 +1,510 @@
+    printf "%s" "$var"
+}
+
+remove_fs_overwrite() {
+    echo 'note: re-running cloud-init will not wipe any LVMs in /dev/md/AUX; to purge these on a re-run, source the metal-lib and invoke enable_fs_overwrite'
+    find /etc/cloud -name "*metalfs.cfg" -exec sed -i 's/overwrite:.*/overwrite: false/g' {} \;
+}
+
+enable_fs_overwrite() {
+    # shellcheck disable=SC2210
+    echo >2 'warning: re-running cloud-init will now wipe all LVMs in /dev/md/AUX!'
+    find /etc/cloud -name "*metalfs.cfg" -exec sed -i 's/overwrite:.*/overwrite: true/g' {} \;
+}
+
+install_grub2() {
+    local working_path
+    local name
+    local index
+    local init_cmdline
+    local disk_cmdline
+    mount -v -L ${boot_authority} -t /etc/fstab.metal 2>/dev/null || echo 'already mounted continuing ...'
+    working_path="$(lsblk -o MOUNTPOINT -nr /dev/disk/by-${boot_scheme,,}/${boot_authority})"
+
+    # Remove all existing entries; anything with CRAY (lower or uppercase). We
+    # only want our boot-loader.
+    for entry in $(efibootmgr | awk -F '[* ]' 'toupper($0) ~ /CRAY/ {print $1}'); do
+         efibootmgr -q -b ${entry:4:8} -B
+    done
+
+    # Install grub2.
+    name=$(grep PRETTY_NAME /etc/*release* | cut -d '=' -f2 | tr -d '"')
+    index=0
+    [ -z "$name" ] && name='CRAY Linux' # Note: if CRAY Linux is observed then the customer has installed a non-SLES distro.
+    for disk in $(mdadm --detail $(blkid -L ${boot_authority}) | grep /dev/sd | awk '{print $NF}'); do
+        # Add '--suse-enable-tpm' to grub2-install once we need TPM.
+        grub2-install --no-rs-codes --suse-force-signed --root-directory $working_path --removable "$disk"
+        efibootmgr -c -D -d "$disk" -p 1 -L "CRAY UEFI OS $index" -l '\efi\boot\bootx64.efi' | grep -i cray
+        index=$((index + 1))
+    done
+
+    # Get the kernel command we used to boot.
+    init_cmdline=$(cat /proc/cmdline)
+    disk_cmdline=''
+    for cmd in $init_cmdline; do
+        # cleans up first argument when running this script on a disk-booted system
+        if [[ $cmd =~ kernel$ ]]; then
+            cmd=$(basename "$(echo $cmd  | awk '{print $1}')")
+        fi
+        if [[ $cmd =~ ^rd.live.overlay.reset ]] ; then :
+        elif [[ $cmd =~ ^rd.debug ]] ; then :
+        # removes all metal vars, and escapes anything that iPXE was escaping
+        # metal vars are used for customizing nodes on deployment, they don't need
+        # to stick around for runtime.
+        # (i.e. ds=nocloud-net;s=http://$url will get the ; escaped)
+        # removes netboot vars
+        elif [[ ! $cmd =~ ^metal. ]] && [[ ! $cmd =~ ^ip=.*:dhcp ]] && [[ ! $cmd =~ ^bootdev= ]]; then
+            disk_cmdline="$(trim $disk_cmdline) ${cmd//;/\\;}"
+        fi
+    done
+
+    # ensure no-wipe is now set for disk-boots.
+    disk_cmdline="$disk_cmdline metal.no-wipe=1"
+
+    # Make our grub.cfg file.
+    # TODO: Add disk-rebuild option
```

Comment:
I don't know if we want to leave TODOs in the scripts that we ship, as it gives the impression that the script is not complete. Todos would be better captured in Jiras than in comments in scripts that are visible to customers, IMO
```suggestion
```

Reply:
It pains me but I'm going to omit amending the metal-lib.sh script with changes to keep the diff to the original script in 1.3/1.4 the same.